src/coreclr/jit/lclvars.cpp

```diff
@@ -3845,6 +3845,57 @@ var_types LclVarDsc::lvaArgType()
     return type;
 }
 
+//------------------------------------------------------------------------
+// GetRegisterType: Determine register type for this local var.
+//
+// Arguments:
+//    tree - node that uses the local, its type is checked first.
+//
+// Return Value:
+//    TYP_UNDEF if the layout is enregistrable, register type otherwise.
+//
+var_types LclVarDsc::GetRegisterType(const GenTreeLclVarCommon* tree) const
+{
+    var_types targetType = tree->gtType;
+
+#ifdef DEBUG
+    // Ensure that lclVar nodes are typed correctly.
+    if (tree->OperIs(GT_STORE_LCL_VAR) && lvNormalizeOnStore())
+    {
+        // TODO: update that assert to work with TypeGet() == TYP_STRUCT case.
```

Comment:
Are you going to fix this "TODO" now, or later? Or should the whole block be removed?

Reply:
fixed